### ISSUE TITLE
Make consistent algorithm description for each sampler and optimiser

### ISSUE THREAD

**Issue by ben18785** [pints-team/pints#672]
I propose that this be of a form like:

- Short description and introduction to the method (usually 2-3 sentences), where we reference any key papers.
- Algorithm pseudocode (as we implement it) written in symbolic form.
- References given in style determined by discussion #671 

How do people feel about this?

**Comment by ben18785** [comment]
Note #673 includes links to inline maths rules for Sphinx docs.

**Comment by MichaelClerx** [comment]
Would be very valuable if we could pull it off, but a few caveats:

- would have to be a long term goal
- we'd have to be very ruthless with leaving out detail, e.g. methods might do lots of clever things beside the main algorithm that we can't cover there
- I really like it when we can make descriptive variable names / method names, rather than `set_epsilon()` etc. This is an opinion not everyone shares I guess. But I'm a bit worried if we have pseudo-code we'll be lazier in the reset of the docs, and then people skimming over methods won't be able to use our code; everyone will just have to read a mathematical summary of every method (only now it's the one we wrote, rather than the original authors'). Or am I being unduly worried?

**Comment by ben18785** [comment]
Yep, agree. In terms of variable names (for use in setters and getters) in many cases there isn't an simple understandable English term for what they represent. Also, in many cases, there is an agreed upon symbol for each variable across the MCMC literature (like epsilon, for example). In these cases, I prefer the Greek letters as think it makes it easier to understand and cross-reference the underlying papers.
